avocado/plugins/nrun.py

```diff
@@ -85,16 +85,30 @@ def spawn_tasks(self, parallel_tasks):
                 print("Finished spawning tasks")
                 break
 
-            yield from self.spawner.spawn_task(task)
+            spawn_result = yield from self.spawner.spawn_task(task)
             identifier = task.identifier
             self.pending_tasks.remove(task)
             self.spawned_tasks.append(identifier)
+            if not spawn_result:
```

Comment:
Just one small comment here. Maybe `spawner.spawn_task` could raise `SpawnerExceptions` with details about what was wrong, so here you could to a `try...except`.

Reply:
It also concerns me that we don't have more detailed information about the failures, and I see that as a further improvement indeed.  I can try to include that in the next (or next next round), which should include scheduler keeping track of those details.  How does that sound?